joomla/joomla-cms
# New field attribute "renderon"

#### Summary of Changes

By means of the new field attribute "renderon" you can make a field dependent upon any global setting or any setting with any component much the same way as the attribute "showon" does for other fields in the same form. The syntax in the "renderon" attribute is identical to the syntax in the "showon" attribute with the following modification: you have to add either the string "global" or the name of a component followed by a colon at the beginning of each check.

You might add this new attribute to any field in any form where this field shall only be shown in this form if one or more global settings and/or component settings have distinct values.
#### Testing Instructions

In order to test this PR you have to change a field definition in any form and make this field dependant upon one or more global settings and/or component settings (no such change is currently part of this PR).

For the following test you need a Joomla installation that is reachable both via HTTP and HTTPS.

A candidate I can suggest is the form in `modules/mod_login/mod_login.xml`. Change the definition of the "usesecure" field as follows:

```
<field
        name="usesecure"
        type="radio"
        label="MOD_LOGIN_FIELD_USESECURE_LABEL"
        description="MOD_LOGIN_FIELD_USESECURE_DESC"
        class="btn-group btn-group-yesno"
        default="0"
        renderon="global:force_ssl:0,1"
        >
        <option value="1">JYES</option>
        <option value="0">JNO</option>
</field>
```

The line starting with "renderon=" is the new line in this definition, I made no other changes here.

Next go to the global configuration and check that the "Force HTTPS" setting is either "None" or "Administrator Only". If the setting is "Entire Site" change it accordingly.

Next go to "Extensions" => "Modules" and search for all site modules containing "login" in their names (with the basic examples installed you should have three such modules).

Open one of these modules and check that the field "Encrypt Login Form" exists (second but last at the "Module" tab). Close this module.

Next go again to the global configuration and set the "Force HTTPS" setting to "Entire Site".

Return to "Extensions" => "Modules" and reopen the module you checked before. The field "Encrypt Login Form" shall no longer exist. Close this module.
